sound/soc/sof/intel/hda-stream.c

```diff
@@ -526,6 +530,7 @@ irqreturn_t hda_dsp_stream_threaded_handler(int irq, void *context)
 			    !s->running ||
 			    (sd_status & SOF_HDA_CL_DMA_SD_INT_MASK) == 0)
 				continue;
+			snd_pcm_period_elapsed(s->substream);
```

Comment:
Ah very interesting. We had a discussion that there should be only one source for period elapsed, but here you are adding another one? Would this work in normal mode w/ the DSP enabled?

Reply:
to use DPIB mode, we need add this and remove the IPC one. Looks the removal of IPC one missed? let me update it.